src/OpenTelemetry.Api/Trace/Tracer.cs

```diff
@@ -147,6 +147,11 @@ public TelemetrySpan WithSpan(TelemetrySpan span)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private TelemetrySpan StartSpanHelper(bool isActiveSpan, string name, SpanKind kind, in SpanContext parentContext = default, SpanAttributes initialAttributes = null, IEnumerable<Link> links = null, DateTimeOffset startTime = default)
         {
+            if (name == null)
```

Comment:
If not specified, could we default to ActivitySource.Name? I feel like Span name used to be set from Tracer name if not specified, so it would be similar. Could be misremembering tho 🐘